src/main/java/com/aws/greengrass/deployment/errorcode/DeploymentErrorCode.java

```diff
@@ -0,0 +1,122 @@
+    RECIPE_PARSE_ERROR(DeploymentErrorType.COMPONENT_RECIPE_ERROR),
+    RECIPE_METADATA_PARSE_ERROR(DeploymentErrorType.COMPONENT_RECIPE_ERROR),
+    ARTIFACT_URI_NOT_VALID(DeploymentErrorType.COMPONENT_RECIPE_ERROR),
+    S3_ARTIFACT_URI_NOT_VALID(DeploymentErrorType.COMPONENT_RECIPE_ERROR),
+    DOCKER_ARTIFACT_URI_NOT_VALID(DeploymentErrorType.COMPONENT_RECIPE_ERROR),
+    EMPTY_ARTIFACT_URI(DeploymentErrorType.COMPONENT_RECIPE_ERROR),
+    EMPTY_ARTIFACT_SCHEME(DeploymentErrorType.COMPONENT_RECIPE_ERROR),
+    UNSUPPORTED_ARTIFACT_SCHEME(DeploymentErrorType.COMPONENT_RECIPE_ERROR),
+    RECIPE_MISSING_MANIFEST(DeploymentErrorType.COMPONENT_RECIPE_ERROR),
+    RECIPE_MISSING_ARTIFACT_HASH_ALGORITHM(DeploymentErrorType.COMPONENT_RECIPE_ERROR),
+    ARTIFACT_CHECKSUM_MISMATCH(DeploymentErrorType.COMPONENT_RECIPE_ERROR),
+    COMPONENT_DEPENDENCY_NOT_VALID(DeploymentErrorType.COMPONENT_RECIPE_ERROR),
+    CONFIG_INTERPOLATE_ERROR(DeploymentErrorType.COMPONENT_RECIPE_ERROR),
+
+    // Config issues
+    DEVICE_CONFIG_NOT_VALID_FOR_ARTIFACT_DOWNLOAD(DeploymentErrorType.DEVICE_ERROR),
+    RUN_WITH_CONFIG_NOT_VALID(DeploymentErrorType.REQUEST_ERROR), UNSUPPORTED_REGION(DeploymentErrorType.REQUEST_ERROR),
+    IOT_CRED_ENDPOINT_FORMAT_NOT_VALID(DeploymentErrorType.REQUEST_ERROR),
+    IOT_DATA_ENDPOINT_FORMAT_NOT_VALID(DeploymentErrorType.REQUEST_ERROR),
+
+    // Docker
+    DOCKER_ERROR(DeploymentErrorType.DEPENDENCY_ERROR), GET_ECR_CREDENTIAL_ERROR(DeploymentErrorType.PERMISSION_ERROR),
+    USER_NOT_AUTHORIZED_FOR_DOCKER(DeploymentErrorType.PERMISSION_ERROR),
+    DOCKER_SERVICE_UNAVAILABLE(DeploymentErrorType.DEPENDENCY_ERROR),
+    DOCKER_LOGIN_ERROR(DeploymentErrorType.DEPENDENCY_ERROR), DOCKER_PULL_ERROR(DeploymentErrorType.DEPENDENCY_ERROR),
+    DOCKER_IMAGE_NOT_VALID(DeploymentErrorType.DEPENDENCY_ERROR),
+
+    // S3
+    S3_ERROR(DeploymentErrorType.DEPENDENCY_ERROR), S3_RESOURCE_NOT_FOUND(DeploymentErrorType.DEPENDENCY_ERROR),
+    S3_ACCESS_DENIED(DeploymentErrorType.PERMISSION_ERROR), S3_BAD_REQUEST(DeploymentErrorType.DEPENDENCY_ERROR),
+    S3_SERVER_ERROR(DeploymentErrorType.SERVER_ERROR),
+    S3_HEAD_OBJECT_ACCESS_DENIED(DeploymentErrorType.PERMISSION_ERROR),
+    S3_HEAD_OBJECT_RESOURCE_NOT_FOUND(DeploymentErrorType.REQUEST_ERROR),
+    S3_GET_BUCKET_LOCATION_ACCESS_DENIED(DeploymentErrorType.PERMISSION_ERROR),
+    S3_GET_BUCKET_LOCATION_RESOURCE_NOT_FOUND(DeploymentErrorType.REQUEST_ERROR),
+    S3_GET_BUCKET_ACCESS_DENIED(DeploymentErrorType.PERMISSION_ERROR),
+    S3_GET_BUCKET_RESOURCE_NOT_FOUND(DeploymentErrorType.REQUEST_ERROR),
+
+    // Nucleus errors
+    EMPTY_DEPLOYMENT_REQUEST(DeploymentErrorType.NUCLEUS_ERROR),
+    DEPLOYMENT_DOCUMENT_PARSE_ERROR(DeploymentErrorType.NUCLEUS_ERROR),
+    DEPLOYMENT_TYPE_NOT_VALID(DeploymentErrorType.NUCLEUS_ERROR),
+    RESOLVE_COMPONENT_CANDIDATES_BAD_RESPONSE(DeploymentErrorType.NUCLEUS_ERROR),
+    COMPONENT_METADATA_NOT_VALID(DeploymentErrorType.NUCLEUS_ERROR),
```

Comment:
does that come from component recipe or is this after downloading we check something and determine it is invalid?

Reply:
also should this be a COMPONENT_ERROR?

Reply:
this is where an invalid component metadata is received in deployment doc. As others, it will be nucleus error if local deployment; cloud service error if cloud deployments